src/commands/fun/compliment.js

```diff
@@ -1,51 +1,30 @@
+import { SlashCommandBuilder, userMention } from "@discordjs/builders";
 import axios from "axios";
-import Commando from "discord.js-commando";
-
-import logger from "../../utils/logger.js";
-
-class ComplimentCommand extends Commando.Command {
-  constructor(client) {
-    super(client, {
-      name: "compliment",
-      group: "fun",
-      memberName: "compliment",
-      description: "Give your dear friends a nice little compliment",
-      argsPromptLimit: 0,
-      args: [
-        {
-          key: "target",
-          prompt: "The user to target",
-          type: "user",
-          validate: undefined,
-          default: "",
-        },
-      ],
-    });
-  }
-
-  async run(message, { target }) {
-    const api = await axios.get(`https://complimentr.com/api`);
-
-    let compliment = api.data.compliment.toLowerCase();
-    compliment = compliment.charAt(0).toUpperCase() + compliment.slice(1);
-
-    logger.info({
-      message: "A compliment was sent...",
-      userId: message.author.id,
-      targetId: target.toString(),
-    });
-
-    if (typeof target === "string") {
-      return await message.say(`Wow, <@${message.author.id}>, ${compliment.toLowerCase()}`);
-    } else {
-      if (target.id === message.author.id) {
-        return await message.reply("Wow, you're really fishing for compliments...");
+import { GuildMember } from "discord.js";
+
+import { SlashCommandHandler } from "../../client";
+
+export default class ComplimentCommand extends SlashCommandHandler {
+  builder = new SlashCommandBuilder()
+    .setName("compliment")
+    .setDescription("Give your dear friends a nice little compliment")
+    .addUserOption((option) => option.setName("user").setDescription("The user to target").setRequired(false));
+
+  handle(interaction, _client) {
+    axios.get(`https://complimentr.com/api`).then((resp) => {
```

Comment:
Kunne vi vært mer konsistente med bruk av `async/await` og `then`? Vil vi helst kun bruker `await` i hele kodebasen, primært for å være konsistente og synes det flyter bedre enn `then`-chains. Her f.eks. returneres det ikke noen verdier, slik som det gjøres i alle de fleste andre commandsene.